networking/src/p2p/peer.rs

```
@@ -315,6 +541,22 @@ impl Receive<SendMessage> for Peer {
     type Msg = PeerMsg;
 
     fn receive(&mut self, ctx: &Context<Self::Msg>, msg: SendMessage, _sender: Sender) {
+        match self.throttle_quota.lock() {
+            Ok(ref mut quota) => {
+                if let Err(quota) = quota.can_send(msg.message.as_ref()) {
+                    if quota == -1 {
+                        warn!(ctx.system.log(), "Cannot send message because of send quota is exceeded";
+                              "msg" => format!("{:?}", msg.message.as_ref()), "peer_uri" => ctx.myself().uri().to_string());
+                    }
+                    return;
+                }
+            }
+            Err(_) => error!(
```

@akoptelov 
do not hide error please, 
```
            Err(e) => error!(
                ctx.system.log(),
                "Failed to obtain a lock on throttling quota";
                "reason" => format!("{:?}", e),
            ),
```

---

@akoptelov 
also change error to warn please

---

@akoptelov 
and what about that `error!` -> `warn!` ?